go-controller/pkg/ovn/egressip.go

```diff
@@ -2205,14 +2209,15 @@ func (e *egressIPController) deletePodEgressIPAssignment(egressIPName string, st
 			metrics.RecordEgressIPUnassign(duration)
 		}()
 	}
-	if err = e.deleteReroutePolicy(podIPs, status, egressIPName); errors.Is(err, libovsdbclient.ErrNotFound) {
+	ops, err := e.deleteReroutePolicyOps(nil, podIPs, status, egressIPName)
+	if errors.Is(err, libovsdbclient.ErrNotFound) {
 		// if the gateway router join IP setup is already gone, then don't count it as error.
 		klog.Warningf("Unable to delete logical router policy, err: %v", err)
 	} else if err != nil {
 		return fmt.Errorf("unable to delete logical router policy, err: %v", err)
 	}
-	var ops []ovsdb.Operation
-	ops, err = deleteNATRuleOps(e.nbClient, []ovsdb.Operation{}, podIPs, status, egressIPName)
+
+	ops, err = deleteNATRuleOps(e.nbClient, ops, podIPs, status, egressIPName)
```

Comment:
I added that to `deletePodEgressIPAssignment` so it is consistent with `addPodEgressIPAssignment`.
I don't think we should do the `deleteEgressIPAssignments` in this PR as it might results in hundreds of ops per txn, this is something we could explore separately.

Reply:
sounds good yes!